clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEvent.java

```diff
@@ -20,4 +20,14 @@
  * This is the abstract definition of the events created by the KafkaConsumer API
  */
 abstract public class ApplicationEvent {
+    public final EventType type;
+
+    public ApplicationEvent(EventType type) {
+        this.type = type;
+    }
+
+    public enum EventType {
+        COMMIT,
+        NOOP,
```

Comment:
Ditto, I think we should keep the testing-related element out of the production code eventually. If we are going to remove it later then that's fine.

Reply:
Thanks, I think the NOOP event will be moved to the test package in the future PR; I left it here to elucidate the purpose of a few methods in the background thread.

Reply:
Thanks for clarifying this, makes sense.